### PR TITLE
Association branch

### PR DESCRIPTION
**Pull request by Gambit142**
- Created a class named `Classroom`
- Created `has-many/belongs-to` relationship between the classes `Classroom` and `Student`
- Created a class named `Book`
- Created a class named `Rental`
- Created a `many-to-many` including a `has-many-through` relationship between classes `Person` and `Book` with the intermediate class `Rentals`

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
+  def initialize(label)
+    @label = label
+    @students = []
```
Comment (TedLivist) [inline classroom.rb]: - [x] Kindly add a getter for the `@students`

File: student.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require './person'
 
 class Student < Person
```
Comment (TedLivist) [inline student.rb:3]: - [ ] You have correctly set up the `has-many` end of the association in the Classroom class. However, your Student class should also have a method for the other side of association setup `(belongs-to)`. You can refer to this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) for an example of how to set up the association both ways.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Book
```
Comment (TedLivist) [inline book.rb:4]: - [ ] Good job setting up the `book.rentals` in the Rental class. We can add a rental to the array with a Rental instance. But this is a `many-to-many` relationship so we should also be able to achieve the `has-many` association from the Book class itself also.

  Kindly create a method (can be named `add_rental`) that takes in a `date` and `person` as parameters and returns an instance of **`Rental`** with `date`, `self`(book) and `person` as its parameters.

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (TedLivist) [inline book.rb]: - [ ] We should be able to modify (set) the `rentals` from the Rental class as you have defined there, but here we can only read (get) the rental. `@rentals` should have a setter and getter, not just getter. Kindly add that.

File: person.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 require_relative './corrector'
 
 class Person
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
```
Comment (TedLivist) [inline person.rb]: - [ ] `:rentals` should have a getter and setter (`:attr_accessor`). Kindly add that.

File: person.rb [hunk 6]
```diff
@@ -1,7 +1,7 @@
 require_relative './corrector'
 
 class Person
```
Comment (TedLivist) [inline person.rb:5]: - [ ] We should also be able to add a rental from the **Person** class. Kindly create a method (can be named `add_rental`) that takes in `book` and `date` as its parameters. The method should return an instance of **`Rental`** with `date`, `book`, and `self` (person) as its parameters

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Gambit142,

Good job so far!
✔️ Neat code
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (Gambit142) [reply classroom.rb]: Thank you for this feedback, I have implemented it.

**Review by mikethreels — APPROVED** [review]
Hi Francis, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gambit142** [comment]
Thank you for the feedback

**Pull request merged**